core-tests/shared/src/test/scala/zio/ZPoolSpec.scala

```diff
@@ -157,14 +157,35 @@ object ZPoolSpec extends ZIOBaseSpec {
           for {
             _ <- ZIO.scoped(ZPool.make(ZIO.unit, 10 to 15, 60.seconds).uninterruptible)
           } yield assertCompletes
-        } +
+        } @@ exceptJS(nonFlaky) +
         test("make preserves interruptibility") {
           val get = ZIO.checkInterruptible(status => ZIO.succeed(status.isInterruptible))
           for {
             pool          <- ZPool.make(get, 10 to 15, 60.seconds)
             _             <- pool.get.repeatN(9)
             interruptible <- pool.get
           } yield assertTrue(interruptible)
-        }
-    }
+        } @@ exceptJS(nonFlaky) +
+        test("doesn't leak resources when acquisition is interrupted") {
+          for {
+            latch     <- Promise.make[Nothing, Unit]
+            latch2    <- Promise.make[Nothing, Unit]
+            incCounter = ZIO.acquireRelease(ZIO.unit)(_ => latch2.succeed(()))
```

Comment:
Note: We can't use a counter in this test because the interruption is "disconnected", so we need to use a promise and await it to check whether it's been completed